shared/common-adapters/index.js

```diff
@@ -1,39 +1,47 @@
 // @flow
 import * as React from 'react'
-import * as I from 'immutable'
 import Box from './box'
 import PopupDialog from './popup-dialog'
+import {connect} from 'react-redux'
 import {isMobile} from '../constants/platform'
 import {globalColors} from '../styles'
 
 const MaybePopup = isMobile
   ? (props: {onClose: () => void, children: React.Node}) => (
       <Box style={{height: '100%', width: '100%'}} children={props.children} />
     )
-  : (props: {onClose: () => void, children: React.Node}) => (
+  : (props: {onClose: () => void, cover?: boolean, children: React.Node}) => (
       <PopupDialog
         onClose={props.onClose}
-        styleCover={_styleCover}
-        styleContainer={_styleContainer}
+        styleCover={props.cover ? _styleCover : {}}
+        styleContainer={props.cover ? _styleContainer : {}}
         children={props.children}
       />
     )
 
-function MaybePopupHoc<P: {}>(WrappedComponent: React.ComponentType<P>) {
-  return (
-    props: P & {navigateUp: Function, routeProps: I.RecordOf<{onClose: (navigateUpFn: Function) => void}>}
-  ) => {
-    const onClose = () => {
-      props.routeProps && props.routeProps.get('onClose') && props.routeProps.get('onClose')(props.navigateUp)
-    }
+// TODO properly type this
+const DispatchNavUpHoc: any = connect(undefined, (dispatch, {navigateUp}) => ({
```

Comment:
I wanted something that would pass down a connectedNavUp so that maybe popup hoc can behave more clearly. It used to be weird in that you'd have to pass a navigateUp function (my fault)